ekncontent/ekncontent/eknc-domain.c

```diff
@@ -730,44 +730,69 @@ eknc_domain_get_fixed_query_finish (EkncDomain *self,
 typedef struct
 {
   // List of EkncContentModel items
-  GSList *models;
+  GPtrArray *models;
   guint total_models;
   guint upper_bound;
+  guint num_received;
 } QueryState;
 
 static void
 query_state_free (gpointer data)
 {
   QueryState *state = data;
-  g_slist_free_full (state->models, g_object_unref);
+  g_clear_pointer (&state->models, g_ptr_array_unref);
```

Comment:
I've done this, adding a function to unref any models which had a ref taken on them by the GPtrArray